pkgs/development/libraries/gdal/default.nix

```diff
@@ -7,13 +7,13 @@ with stdenv.lib;
 
 stdenv.mkDerivation rec {
   pname = "gdal";
-  version = "3.0.3";
+  version = "3.0.4";
 
   src = fetchFromGitHub {
     owner = "OSGeo";
     repo = "gdal";
     rev = "v${version}";
-    sha256 = "1rbyxmgmp27a5wvm4g70jr79bazhdl8q9rcch2b78m73njdv73xa";
+    sha256 = "00a7q9wv8s1bmdhqxvixkq2afr8aibg3pkc76gg50r8lavf6j84c";
```

Comment:
would you mind adding the following change to fix the build on darwin?
```diff
diff --git c/pkgs/development/libraries/gdal/default.nix i/pkgs/development/libraries/gdal/default.nix
index 7f3e796bbf1..0c16b2de240 100644
--- c/pkgs/development/libraries/gdal/default.nix
+++ i/pkgs/development/libraries/gdal/default.nix
@@ -13,7 +13,12 @@ stdenv.mkDerivation rec {
     owner = "OSGeo";
     repo = "gdal";
     rev = "v${version}";
-    sha256 = "00a7q9wv8s1bmdhqxvixkq2afr8aibg3pkc76gg50r8lavf6j84c";
+    sha256 = "0y33fi96bmkbnrgcgfg5ih0hw44k7093x4n8d81cbgcnz0b1hj7w";
+    # Remove unicode file names which leads to different checksums on HFS+
+    # vs. other filesystems because of unicode normalisation.
+    extraPostFetch = ''
+      rm -r $out/autotest/pyscripts/data/
+    '';
   };
 
   sourceRoot = "source/gdal";
```

Reply:
@marsam Thanks for testing on darwin!

I'm wondering if we can get an upstream fix for this issue. GDAL looks like a very active, well-maintained project that is tested on numerous platforms including MacOS X. It's possible there's already an upstream commit that addresses this issue; there have been 20 non-merge commits on the release/3.0 branch since v3.0.4 was released on Jan 28.

Also, from hydra, it looks like v3.0.3 was building successfully for darwin and there were only 15 non-merge commits between v3.0.3 (tagged Jan 8, 2020) and v3.0.4 (tagged Jan 28, 2020). So, if it hasn't already been fixed, it might not be too hard to bisect and report the issue. I would be happy to do this, but I do not have access to a darwin system.

Perhaps OfBorg would be of some help here, although I do not currently have access.

What do you recommend?

Reply:
Unfortunately it's an issue in HFS+ which does unicode normalisation, which is not the case in other filesystems commonly found in Linux https://github.com/NixOS/nix/issues/847. 
>there were only 15 non-merge commits between v3.0.3 (tagged Jan 8, 2020) and v3.0.4 (tagged Jan 28, 2020)

Curiously, `v3.0.3` also yields a different hash. I guess the hydra builders share a cache of the sources, and the darwin builder finding the hash already in the cache, didn't calculate it again.

Since it's an issue in _our_ side, I'm inclined to patch it here

Reply:
Thanks for the clarification.

It seems unfortunate to lose the benefit of a set of tests across all platforms, especially since the test file names do not all contain unicode characters -- and new test files might get added in future GDAL releases.

Is there any way to make the patch platform-dependent?

Reply:
As a follow-up, here are some related questions (not sure the best way to answer):

1. Does the HFS+ unicode issue still exist with Apple's newer APFS filesystem? It seems to handle unicode differently. See https://en.wikipedia.org/wiki/Apple_File_System.

2. Are the hydra builders for darwin configured with HFS+ or APFS?

3. Assuming the darwin builders access a shared source cache (as suggested above), would the packages be flagged as corrupt by a nix verify on the darwin builders?

I'm just trying to understand the issue a little better. It seems that the workaround of deleting files with unicode filenames only appears in a few packages. I'm wondering if unicode filenames with normalization issues are exceedingly rare, or the issue is being masked some other way.

Reply:
@marsam After some research, it seems that there are at least 30 sources fetched from github that include unicode filenames and which are required by darwin packages. Only six currently have an extraPostFetch to remove unicode files.

It's possible that some of the unicode filenames in the other packages end-up hashing "correctly" on darwin. But, maybe the issue is being masked, as you suggest, by a shared source cache. Thus, a hash mismatch is only identified when a darwin user builds one of these packages locally.

Perhaps a separate pull request could be opened to address packages that do not hash correctly on darwin due to unicode filenames (https://github.com/NixOS/nix/issues/847), e.g. "Fix unicode file handling for darwin". This PR could be used to fix/track all affected packages uniformly. Since the number of affected packages seems to be increasing, the extra visibility might lead to a more general solution.

In the meantime, to avoid conflating issues, I think it makes sense to keep this PR as a simple revision update.

Reply:
Sorry, I don't have much information to answer your questions about the darwin builders, but if you are curious maybe you can try in the #nixos irc channel

>I think it makes sense to keep this PR as a simple revision update

Sure, no problem